### PR TITLE
Frontend Libs 9.2.0

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
### Added
- 12 new UI icons
- 1 new block icon
- new `lodash` function replacements
- `ColorPicker` has new configuration options available: `expanded`, `border`, `inlineLabel`
- `OptionSelector` has a new `labelOnlyOnActive` configuration option
- `PopoverWithTrigger` got a new prop injection functionality to simplify external popover closing (via `allowCloseFromChildren` on the Popover, and `esClosesModalOnClick` on any element child)
- `.es-tabular-nums`, `.es-hidden` (`visibility: hidden`) and `es-dots-background` (dotted background) utility classes have been added

### Updated
- Project will now only lint staged files
- All of the dependencies have been updated to the latest versions
- Stylelint rules have been cleaned up, and previous rules have been reintroduced from a different package, due to changes in Stylelint
- Updated some of the current helpers to use new, lightweight implementations
- Fixed some linter errors
- Fixed a bug within Quote block where an attribute was missing
- `AdvancedColorPicker` has a new design
- Fixed a bug with gradient picker inside `AdvancedColorPicker`
- `ColorPalette` has a new design
- `ColorPicker` has a refreshed design
- `ColorSwatch` has a new design and its performance was optimized a bit
- `ColorSwatch` has a new swatch for "undefined" colors, so the empty state looks nicer (if `es-undefined` is passed as a color value)
- Heading, Paragraph, Video, and Button blocks/components have had their options tweaked
- Fixed wrong gap in `UseToggle` with the only toggle and label visible
- `getPaletteColors()` helper now returns from the built-in FE libs stores, instead of WP colors
- Fixed tests

### Removed
- `lodash` is no longer used in the project

Refreshed _Button_ & _Heading_
image
image

New `ColorPicker` design
image
image

New `AdvancedColorPicker`
image
image
image

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (piqusy) [inline .husky/pre-commit:4]: 💪🏼

File: blocks/init/src/Blocks/components/heading/components/heading-options.js [hunk 2]
```diff
@@ -47,13 +45,22 @@ export const HeadingOptions = (attributes) => {
...
+		{ label: 'H4', tooltip: __('Heading 4', 'eightshift-frontend-libs'), value: 4 },
+		{ label: 'H5', tooltip: __('Heading 5', 'eightshift-frontend-libs'), value: 5 },
+		{ label: 'H6', tooltip: __('Heading 6', 'eightshift-frontend-libs'), value: 6 },
+	];
+
 	return (
 		<UseToggle {...generateUseToggleConfig(attributes, manifest, 'headingUse')}>
 			<Section
-				showIf={!hideColor || !hideSize || !hideFontWeight}
+				showIf={!hideColor || !hideSize || !hideFontWeight || !hideFontWeight}
```
Comment (piqusy) [inline blocks/init/src/Blocks/components/heading/components/heading-options.js:60]:
```suggestion
				showIf={!hideColor || !hideSize || !hideFontWeight || !hideFontWeight}
```
Double trouble

File: scripts/components/fancy-divider/fancy-divider.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ export const FancyDivider = (props) => {
 				additionalClasses='es-nested-bg-cool-gray-450 es-nested-rounded-1 es-nested-color-pure-white! es-has-enhanced-contrast-icon'
 			/>
 
-			<div className='es-w-full es-flex-1 es-h-px es-bg-cool-gray-100'></div>
+			<div className='es-w-full es-flex-1 es-h-px es-bg-cool-gray-100' />
```
Comment (piqusy) [inline scripts/components/fancy-divider/fancy-divider.js:42]: All previous changes in this manner have this written as the full tag, why the exception here?

File: scripts/components/option-selector/option-selector.js [hunk 4]
```diff
@@ -103,7 +105,7 @@ export const OptionSelector = (props) => {
 
 				if (iconData && typeof iconData === 'string' && (iconData?.startsWith('<svg') ?? false)) {
 					icon = (
-						<i className='es-line-h-0 es-display-contents' dangerouslySetInnerHTML={{ __html: iconData }}></i>
+						<i className='es-line-h-0 es-display-contents' dangerouslySetInnerHTML={{ __html: iconData }} />
```
Comment (piqusy) [inline scripts/components/option-selector/option-selector.js:108]: Same as above (why the shorthand now?)

File: scripts/editor/registration.js [hunk 5]
```diff
@@ -509,14 +509,14 @@ export const getIconOptions = (
 		return {
 			background: (typeof icon.background === 'undefined') ? backgroundGlobal : icon.background,
 			foreground: (typeof icon.foreground === 'undefined') ? foregroundGlobal : icon.foreground,
-			src: <span dangerouslySetInnerHTML={{ __html: blockIcons[icon.src] }}></span>,
+			src: <span dangerouslySetInnerHTML={{ __html: blockIcons[icon.src] }} />,
```
Comment (piqusy) [inline scripts/editor/registration.js:512]: Shorthand?

File: scripts/editor/registration.js [hunk 6]
```diff
@@ -509,14 +509,14 @@ export const getIconOptions = (
...
 		};
 	}
 
 	return {
 		background: (typeof icon.background === 'undefined') ? backgroundGlobal : icon.background,
 		foreground: (typeof icon.foreground === 'undefined') ? foregroundGlobal : icon.foreground,
-		src: icon.src.includes('<svg') ? <span dangerouslySetInnerHTML={{ __html: icon.src }}></span> : icon.src,
+		src: icon.src.includes('<svg') ? <span dangerouslySetInnerHTML={{ __html: icon.src }} /> : icon.src,
```
Comment (piqusy) [inline scripts/editor/registration.js:519]: Shorthand?

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few comments

**Review by dadadavorin — DISMISSED** [review]
Nice work! 👏 💪

File: scripts/components/fancy-divider/fancy-divider.js (hunk 3 above)
Comment (iruzevic) [reply scripts/components/fancy-divider/fancy-divider.js:42]: in jsx you can write this this way it doesn't really mater

**Review by iruzevic — DISMISSED** [review]

File: scripts/components/fancy-divider/fancy-divider.js (hunk 3 above)
Comment (piqusy) [reply scripts/components/fancy-divider/fancy-divider.js:42]: I know, but in some places in this PR this approach has reversed logic... just wondering why the inconsistencies 😄

**Review by kancijan — DISMISSED** [review]

File: scripts/components/fancy-divider/fancy-divider.js (hunk 3 above)
Comment (goranalkovic-infinum) [reply scripts/components/fancy-divider/fancy-divider.js:42]: It's an linter error now 😄
"Empty components are self-closing"

File: composer.json [hunk 7]
```diff
@@ -46,7 +46,7 @@
 	},
 	"scripts": {
 		"test:types": "@php ./vendor/bin/phpstan analyze",
-		"test:standards": "@php ./vendor/squizlabs/php_codesniffer/bin/phpcs --runtime-set testVersion 7.4-",
+		"test:standards": "@php ./vendor/squizlabs/php_codesniffer/bin/phpcs --runtime-set testVersion 8.2-",
```
Comment (dadadavorin) [inline composer.json:49]: 👌

**Review by dadadavorin — DISMISSED** [review]

**Review by piqusy — DISMISSED** [review]

**Review by vedranpetrac — DISMISSED** [review]

**Review by iobrado — DISMISSED** [review]

**Review by dadadavorin — APPROVED** [review]

**Review by iruzevic — APPROVED** [review]

**Pull request merged**